src/request-data/form.js

```diff
@@ -66,7 +66,28 @@ export default () => {
   });
 
   const formDatasetDiff = createResource('formDatasetDiff');
-  const formDraftDatasetDiff = createResource('formDraftDatasetDiff');
+  const formDraftDatasetDiff = createResource('formDraftDatasetDiff', () => ({
+    counts: computeIfExists(() => {
+      let updatedDatasets = 0;
+      let newProperties = 0;
+      for (const dataset of formDraftDatasetDiff) {
+        if (dataset.isNew) {
+          updatedDatasets += 1;
+          newProperties += dataset.properties.length;
```

Comment:
Add test for the case where there's a new dataset without any properties (only the label will be updated).